File: hackerrank/easy/0035-append-and-delete/js/solution.mjs
// solution.mjs

export function appendAndDelete(s, t, k) {
  if (k >= s.length + t.length) return "Yes";

  let common = 0;
  for (let i = 0; i < Math.min(s.length, t.length); i++) {
    if (s[i] === t[i]) common++;
    else break;
  }

  const deletions = s.length - common;
  const additions = t.length - common;
  const totalChanges = deletions + additions;

  return k >= totalChanges && (k - totalChanges) % 2 === 0 ? "Yes" : "No";
}

// --- demo runs ---
console.log(appendAndDelete("hackerhappy", "hackerrank", 9)); // Yes
console.log(appendAndDelete("aba", "aba", 7)); // Yes
console.log(appendAndDelete("ashley", "ash", 2)); // No

File: hackerrank/easy/0035-append-and-delete/swift/solution.swift
// solution.swift

func appendAndDelete(s: String, t: String, k: Int) -> String {
    // If we can remove and rebuild both entirely
    if k >= (s.count + t.count) {
        return "Yes"
    }

    let sArr = Array(s)
    let tArr = Array(t)
    let minLen = min(sArr.count, tArr.count)
    var common = 0

    for i in 0..<minLen {
        if sArr[i] == tArr[i] {
            common += 1
        } else {
            break
        }
    }

    let deletions = sArr.count - common
    let additions = tArr.count - common
    let totalChanges = deletions + additions

    if k >= totalChanges && (k - totalChanges) % 2 == 0 {
        return "Yes"
    } else {
        return "No"
    }
}

// --- demo runs ---
print(appendAndDelete(s: "hackerhappy", t: "hackerrank", k: 9)) // Yes
print(appendAndDelete(s: "aba", t: "aba", k: 7)) // Yes
print(appendAndDelete(s: "ashley", t: "ash", k: 2)) // No